### PR TITLE
Updated documentation to reflect edX changes

### LINKED ISSUE #450: Update documentation to reflect edX changes
We need to update the README to reflect 2 main changes that edX has made:

1. The structure (and content) of their docs for running edX devstack has been changed. Links to those docs need to be updated, and the instructions on our end may need to be updated as well.
1. The process for adding OAuth credentials has changed. We now need to add an Application in "Django OAuth toolkit" (with the "Authorization code" auth type). The README needs to be updated to reflect this

We should also document the steps for unlocking mongo on edX devstack

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?

Closes #450 

#### What's this PR do?

Updates our README to reflect updated edX docs, the new auth process, and a couple other minor details

#### How should this be manually tested?

If you feel particularly enterprising, you can run through the entire README to confirm the accuracy of these instructions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,16 +88,17 @@ need to use Django admin for the next step anyway.
...
-- **Name**: Anything you want. Something like 'micromasters-local'
-- **Url**: The URL where MicroMaster’s will be running. If you're running it via
-Docker, run ``docker-machine ip`` from the host machine to get the container IP.
-MicroMaster’s runs on port ``8079`` by default, so this value should be something
-like ``http://192.168.99.100:8079``
-- **Redirect uri**: Your **Url** value with "/complete/edxorg/" at the end
+- **Redirect uris**: The URL where MicroMaster’s will be running, followed by "/complete/edxorg/".
+If you're running it via Docker, run ``docker-machine ip`` from the host machine to get the
+container IP. MicroMaster’s runs on port ``8079`` by default, so this value should be something
+like ``http://192.168.99.100:8079/complete/edxorg/``
```
Comment (pdpinch) [inline README.md]: I believe this IP is only correct if you're on a Mac and using docker-machine?

@aliceriot @noisecapella do you use localhost for the redirect uri?

File: README.md [hunk 2]
```diff
@@ -101,10 +112,28 @@ a template to create your ``.env`` file. For MicroMaster’s to work, it needs 3
...
-    your new client. Use those values for the corresponding ``EDXORG_``
-    variables in the ``.env`` file
+    that new Application. Use those values for the corresponding ``EDXORG_``
+    variables in the ``.env`` file.
+
+#### General edX devstack debugging notes
+
+- To update your devstack with important changes from edX, run `vagrant provision` in
+your edx_devstack directory. This will pull down the latest release and run migrations, among
+other things.
```
Comment (pdpinch) [inline README.md]: Will running `vagrant provision` on a freshly installed devstack pull the current master, or something else?

**Comment by pdpinch** [comment]
A couple of questions. Only the first one needs to be answered in these docs, I think.

File: README.md (hunk 2 above)
Comment (pdpinch) [inline README.md]: In my tests today, `vagrant provision` does a git fetch and switches to the release branch, so the text can stay as is here.

File: README.md (hunk 1 above)
Comment (aliceriot) [inline README.md]: I use http://localhost:8079/complete/edxorg/ for the redirect uri on micromasters.d.

**Comment by gsidebo** [comment]
@pdpinch ready to review again. i shoe-horned in an addition to .gitignore. we never check in the `release-notes-checklist` file when we run a release

File: README.md [hunk 3]
```diff
@@ -93,9 +93,11 @@ new Application. Fill in the values as follows:
 
 - **User**: Use the lookup (magnifying glass) to find your superuser
 - **Redirect uris**: The URL where MicroMaster’s will be running, followed by "/complete/edxorg/".
-If you're running it via Docker, run ``docker-machine ip`` from the host machine to get the
-container IP. MicroMaster’s runs on port ``8079`` by default, so this value should be something
-like ``http://192.168.99.100:8079/complete/edxorg/``
+ **Linux users:** the MicroMaster’s URL will be `localhost:8079`. **OSX users:** In a terminal window on the
+ host machine where the proper Docker environment has been set up (`eval "$(docker-machine env mm)"`), the
+ MicroMaster's IP can be found by running ``docker-machine ip`` from the host machine. MicroMaster’s runs on
```
Comment (giocalitri) [inline README.md]: you need to specify the name of the machine, so it is `docker-machine ip <machine_name>`

Comment (gsidebo) [inline README.md]: fixed

Comment (giocalitri) [inline README.md]: I still do not see the <machine_name> (or whatever you want to use)

Comment (gsidebo) [inline README.md]: fixed again :)

**Comment by giocalitri** [comment]
👍

**Pull request merged**